File: src/main/java/com/aitherapist/aitherapist/scheduled/ScheduledNotification.java
package com.aitherapist.aitherapist.scheduled;

import com.aitherapist.aitherapist.domain.enums.NotificationStatus;
import jakarta.persistence.*;
import java.time.LocalDateTime;

@Entity
@Table(name = "scheduled_notifications")
public class ScheduledNotification {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "internal_user_id")
    private Long internalUserId;

    @Column(name = "telegram_chat_id", nullable = false)
    private Long telegramChatId;

    @Column(nullable = false, columnDefinition = "TEXT")
    private String message;

    @Column(name = "trigger_time", nullable = false)
    private LocalDateTime triggerTime;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private NotificationStatus status = NotificationStatus.PENDING;

    @Column(name = "created_at", nullable = false)
    private LocalDateTime createdAt = LocalDateTime.now();

    public ScheduledNotification() {
    }

    public ScheduledNotification(Long id, Long internalUserId, Long telegramChatId,
                                 String message, LocalDateTime triggerTime,
                                 NotificationStatus status, LocalDateTime createdAt) {
        this.id = id;
        this.internalUserId = internalUserId;
        this.telegramChatId = telegramChatId;
        this.message = message;
        this.triggerTime = triggerTime;
        this.status = status;
        this.createdAt = createdAt;
    }

    // Getters and Setters
    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public Long getInternalUserId() {
        return internalUserId;
    }

    public void setInternalUserId(Long internalUserId) {
        this.internalUserId = internalUserId;
    }

    public Long getTelegramChatId() {
        return telegramChatId;
    }

    public void setTelegramChatId(Long telegramChatId) {
        this.telegramChatId = telegramChatId;
    }

    public String getMessage() {
        return message;
    }

    public void setMessage(String message) {
        this.message = message;
    }

    public LocalDateTime getTriggerTime() {
        return triggerTime;
    }

    public void setTriggerTime(LocalDateTime triggerTime) {
        this.triggerTime = triggerTime;
    }

    public NotificationStatus getStatus() {
        return status;
    }

    public void setStatus(NotificationStatus status) {
        this.status = status;
    }

    public LocalDateTime getCreatedAt() {
        return createdAt;
    }

    public void setCreatedAt(LocalDateTime createdAt) {
        this.createdAt = createdAt;
    }
}

File: src/main/java/com/aitherapist/aitherapist/scheduled/ScheduledNotification.kt
package com.aitherapist.aitherapist.telegrambot.scheduled

import com.aitherapist.aitherapist.domain.enums.NotificationStatus
import jakarta.persistence.*
import java.time.LocalDateTime

/**
 * ScheduledNotification - Entity class. transfer notification message to user.
 * use hibernate .
 *
 */
@Entity
@Table(name = "scheduled_notifications")
class ScheduledNotification(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long = 0,

    @Column(name = "internal_user_id")
    val internalUserId: Long? = null,

    @Column(name = "telegram_chat_id", nullable = false)
    val telegramChatId: Long,

    @Column(nullable = false, columnDefinition = "TEXT")
    val message: String,

    @Column(name = "trigger_time", nullable = false)
    val triggerTime: LocalDateTime,

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    var status: NotificationStatus = NotificationStatus.PENDING,

    @Column(name = "created_at", nullable = false)
    val createdAt: LocalDateTime = LocalDateTime.now()
)
